lesson10: Eugene.Ovdiyuk

ovdiyuk opened the pull request:
Implemented RT_MUTEX, SPINLOCK, SEMAPHORE in the one commit
RT_MUTEX is active in the code;

Please uncomment these to test:
DEFINE_SPINLOCK(spinlockname)
DEFINE_SEMAPHORE(semaphorename);
and related methods in the code.

ovdiyuk committed:
10-ConcurrencyAndSynchronization: Initial commit

Implemented RT_MUTEX, SPINLOCK, SEMAPHORE

IevgenMaliarenko commented on 10-ConcurrencyAndSynchronization/my_thread.c:
```diff
@@ -0,0 +1,72 @@
+#include <linux/time.h>
+#include <linux/timer.h>
+#include <linux/delay.h>
+#include <linux/spinlock.h>
```
unused headers should not be included

ovdiyuk replied:
Fixed. Thanks

IevgenMaliarenko commented on 10-ConcurrencyAndSynchronization/my_thread.c:
```diff
@@ -0,0 +1,72 @@
+
+//Please use one single RT_MUTEX or SPINLOCK or SEMAPHORE
+static DEFINE_RT_MUTEX(mutexname);
+// static DEFINE_SPINLOCK(spinlockname);
```
try to create commit history instead of commented code

ovdiyuk replied:
Sorry,  but I made this way to have these examples in my code for future uses.
I don't wish to used different commits for different thread implementation approaches.
Please let me leave this as is to have examples on single IDE page in my branch.

IevgenMaliarenko commented on 10-ConcurrencyAndSynchronization/my_thread.c:
```diff
@@ -0,0 +1,72 @@
+module_exit(mymodule_exit);
+ 
+MODULE_AUTHOR("Eugene.Ovdiyuk <ovdiyuk@gmail.com>");
+MODULE_DESCRIPTION("Simple sysfs module");
```
it's not a simple sysfs module

ovdiyuk replied:
Fixed. Thanks.

IevgenMaliarenko reviewed (commented):
What is results or conclusions regarding your investigation, I mean rt-mutex/spilock usage?

ovdiyuk committed:
10-ConcurrencyAndSynchronization: Minor fixes

1. MODULE_DESCRIPTION: Renamed Module to application
2. Updated functions names
3. Deleted unused headers

ovdiyuk commented:
About the results of the investigation.
I was able to run more than 4 threads with rt-mutex, but it's not possible with spilock usage. Not sure in the reason?
